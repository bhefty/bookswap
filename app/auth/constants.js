export const LOGIN_REQUEST = 'boilerplate/auth/LOGIN_REQUEST'
export const LOGIN_SUCCESS = 'boilerplate/auth/LOGIN_SUCCESS'
export const LOGIN_FAILURE = 'boilerplate/auth/LOGIN_FAILURE'
export const LOGOUT = 'boilerplate/auth/LOGOUT'
