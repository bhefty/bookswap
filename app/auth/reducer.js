import { fromJS } from 'immutable'

// import { getStoredAuthState } from 'utils/auth'

import {
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
  LOGOUT
} from './constants'

const initialState = fromJS({
  isLoggingIn: false,
  idToken: null,
  profile: null,
  error: null
})

function authReducer (state = initialState, action) {
  switch (action.type) {
    case LOGIN_REQUEST:
      return state
        .set('isLoggingIn', true)
    case LOGIN_SUCCESS:
      return state
        .set('isLoggingIn', false)
        .set('idToken', action.idToken)
        .set('profile', action.profile)
    case LOGIN_FAILURE:
      return state
        .set('isLogginIn', false)
        .set('idToken', null)
        .set('profile', null)
        .set('error', action.error)
    case LOGOUT:
      return initialState
    default:
      return state
  }
}

export default authReducer
