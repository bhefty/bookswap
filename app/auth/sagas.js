import Auth0Lock from 'auth0-lock'
import { push } from 'react-router-redux'
import { call, put, take } from 'redux-saga/effects'

import { removeStoredAuthState, setStoredAuthState } from 'utils/auth'

import {
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
  LOGOUT
} from './constants'

import { loginSuccess, loginFailure } from './actions'

export function * loginRequestSaga () {
  const lock = new Auth0Lock(
    process.env.AUTH0_CLIENT_ID,
    process.env.AUTH0_DOMAIN,
    {
      auth: { redirect: false },
      languageDictionary: { title: 'React Redux Auth Boilerplate' }
    }
  )

  const showLock = () => {
    new Promise((resolve, reject) => { // eslint-disable-line no-new
      lock.on('hide', () => reject('Lock closed')) // eslint-disable-line prefer-promise-reject-errors

      lock.on('authenticated', (authResult) => {
        lock.getuserInfo(authResult.accessToken, (error, profile) => {
          if (!error) {
            lock.hide()
            resolve({ profile, idToken: authResult.idToken })
          }
        })
      })

      lock.on('unrecoverable_error', (error) => {
        lock.hide()
        reject(error)
      })

      lock.show()
    })
  }

  try {
    const { profile, idToken } = yield call(showLock)

    yield put(loginSuccess(profile, idToken))
    yield put(push('/dashboard'))
  } catch (error) {
    yield put(loginFailure(error))
    yield put(put('/'))
  }
}

export function * watchLoginRequest () {
  while (true) {
    yield take(LOGIN_REQUEST)
    yield call(loginRequestSaga)
  }
}

export function * watchLoginSuccess () {
  while (true) {
    const { profile, idToken } = yield take(LOGIN_SUCCESS)

    setStoredAuthState(profile, idToken)
  }
}

export function * watchLoginFailure () {
  while (true) {
    yield take(LOGIN_FAILURE)

    removeStoredAuthState()
  }
}

export function * watchLogout () {
  while (true) {
    yield take(LOGOUT)

    removeStoredAuthState()

    yield put(push('/'))
  }
}

export function * authSagas () {
  return [
    watchLoginSuccess,
    watchLoginRequest,
    watchLoginFailure,
    watchLogout
  ]
}

// export default function * authSaga () {
//   yield [
//     fork(watchLoginSuccess),
//     fork(watchLoginRequest),
//     fork(watchLoginFailure),
//     fork(watchLogout)
//   ]
// }
