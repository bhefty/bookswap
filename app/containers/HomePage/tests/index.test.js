import React from 'react'
import { shallow, mount } from 'enzyme'

import { HomePage, mapDispatchToProps } from '../index'
import { loginRequest } from 'auth/actions'

describe('<HomePage />', () => {
  it('should render the page', () => {
    const renderedComponent = mount(<HomePage />)
    expect(renderedComponent.exists())
  })

  it('should render Splash component', () => {
    const renderedComponent = shallow(
      <HomePage />
    )
    expect(renderedComponent.find('Splash').length).toEqual(1)
  })

  describe('mapDispatchToProps', () => {
    describe('login', () => {
      it('should be injected', () => {
        const dispatch = jest.fn()
        const result = mapDispatchToProps(dispatch)
        expect(result.login).toBeDefined()
      })

      it('should dispatch login when called', () => {
        const dispatch = jest.fn()
        const result = mapDispatchToProps(dispatch)
        result.login()
        expect(dispatch).toHaveBeenCalledWith(loginRequest())
      })
    })
  })
})
