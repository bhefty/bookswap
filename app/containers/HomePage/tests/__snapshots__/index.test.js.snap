// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render the page 1`] = `
<div>
  <h1>
    Welcome!
  </h1>
  Current counter value: 
  <div>
    <button
      onClick={undefined}
    >
      Increment
    </button>
  </div>
  <div>
    <button
      onClick={undefined}
    >
      Decrement
    </button>
  </div>
</div>
`;
